libraries/hatch-server-user-management/src/AWSCognitoClient.ts

```diff
@@ -48,39 +46,49 @@ export default class AWSCognitoClient implements UserServiceClient {
   public async startUserRegistration(username: string, password: string, userAttributes?: {[key: string]: any}) {
     let cognitoUserAttributes: AttributeListType = [];
     for (const key in userAttributes) {
-      cognitoUserAttributes.push({
-        Name: key,
-        Value: userAttributes[key],
-      });
-    };
+      if (userAttributes) {
```

Comment:
Should be outside the for loop. We should also probably be iterating over Object.keys or similar.

Reply:
updated